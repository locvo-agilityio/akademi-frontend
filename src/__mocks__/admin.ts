export const MOCK_ADMIN = {
  id: 1,
  name: 'Nabila A',
  role: 'Admin',
  avatar: 'https://bit.ly/dan-abramov',
};
