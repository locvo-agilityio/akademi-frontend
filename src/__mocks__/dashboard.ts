import {
  CalendarIcon,
  FoodIcon,
  StudentIcon,
  TeacherIcon,
} from '@/components/icons';

export const MOCK_SUMMARY_DASHBOARD = [
  {
    title: 'Students',
    total: 932,
    bgIcon: 'primary',
    icon: StudentIcon,
  },
  {
    title: 'Teachers',
    total: 754,
    bgIcon: 'secondary',
    icon: TeacherIcon,
  },
  {
    title: 'Events',
    total: 40,
    bgIcon: 'yellow',
    icon: CalendarIcon,
  },
  {
    title: 'Foods',
    total: 32,
    bgIcon: 'darkBlue',
    icon: FoodIcon,
  },
];
