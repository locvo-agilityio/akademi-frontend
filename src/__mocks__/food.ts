export const MOCK_FOODS = [
  {
    name: 'Beef Steak with Fried Potato',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    image:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRCQPUcFz7CxVScB-zxL5pl2w6f4ntjw4mV-w&s',
  },
  {
    name: 'Beef Steak with Fried Potato New',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    image:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRCQPUcFz7CxVScB-zxL5pl2w6f4ntjw4mV-w&s',
  },
];
