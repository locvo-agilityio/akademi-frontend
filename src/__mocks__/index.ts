export * from './admin';
export * from './dashboard';
export * from './student';
export * from './pagination';
export * from './message';
export * from './food';
export * from './teacher';
export * from './schedule';
export * from './payment';
