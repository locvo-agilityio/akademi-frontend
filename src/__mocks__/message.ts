export const MOCK_MESSAGES = [
  {
    name: 'Samantha William',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
  {
    name: 'Tony Soap',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
  {
    name: 'Jordan Nico',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
  {
    name: 'Nadila Adja',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    avatar: 'https://bit.ly/dan-abramov',
    time: '12:45 PM',
  },
];
