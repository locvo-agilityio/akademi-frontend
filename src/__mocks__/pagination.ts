export const MOCK_PAGINATION = {
  totalRecords: '100 items',
  pageSize: 10,
  currentButtons: [1, 2, 3],
  currentPage: 1,
};
