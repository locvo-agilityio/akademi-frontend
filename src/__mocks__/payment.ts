export const MOCK_PAYMENTS = [
  {
    id: 1,
    date: String(new Date()),
    amount: 100000,
    status: 'Pending',
  },
  {
    id: 2,
    date: String(new Date()),
    amount: 100000,
    status: 'Complete',
  },
  {
    id: 3,
    date: String(new Date()),
    amount: 100000,
    status: 'Canceled',
  },
  {
    id: 4,
    date: String(new Date()),
    amount: 100000,
    status: 'Complete',
  },
];
