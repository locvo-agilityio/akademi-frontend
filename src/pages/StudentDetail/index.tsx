'use client';

import { lazy, Suspense } from 'react';
import { useParams } from 'react-router-dom';
import { Box, HStack } from '@chakra-ui/react';

// Hooks
import { useGetStudent } from '@/hooks';

// UI
import { TablePayment } from '@/ui';

// Components
import { Fallback } from '@/components';

const Profile = lazy(() => import('@/ui/StudentDetail/Profile'));
const Schedule = lazy(() => import('@/ui/StudentDetail/Schedule'));

const StudentDetail = () => {
  const id = useParams().id || '';
  const { student } = useGetStudent(id);
  const { parentName, address, email, phone, schedule, avatar } =
    student?.data || {};

  return (
    <HStack
      gap={10}
      mt={5}
      alignItems="flex-start"
      justifyContent="space-between"
    >
      <Box w="full">
        <Suspense fallback={<Fallback />}>
          <Profile
            parentName={parentName}
            address={address}
            email={email}
            phone={phone}
            avatar={avatar}
          />
        </Suspense>

        <TablePayment />
      </Box>

      <Suspense fallback={<Fallback />}>
        <Schedule schedules={schedule} />
      </Suspense>
    </HStack>
  );
};

export default StudentDetail;
