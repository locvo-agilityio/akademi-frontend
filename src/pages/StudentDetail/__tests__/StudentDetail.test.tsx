import { render } from '@testing-library/react';

// Components
import StudentDetail from '..';

// Mocks
import { MOCK_TEACHERS, MOCK_UNPAID_STUDENTS } from '@/__mocks__';

jest.mock('@/hooks', () => ({
  useGetStudent: jest.fn(() => ({
    student: MOCK_TEACHERS[0],
  })),
  useGetUnpaidStudents: jest.fn(() => ({
    unpaidStudents: {
      data: MOCK_UNPAID_STUDENTS,
      meta: {
        pagination: {
          page: 1,
          pageSize: 10,
          pageCount: 3,
          total: 30,
        },
      },
    },
    isUnpaidStudentsLoading: false,
  })),
  usePagination: jest.fn(() => ({
    pageArray: [1, 2],
    isDisableNext: false,
    isDisablePrev: true,
    handleChangePageNumber: jest.fn(),
    handleChangePage: jest.fn(),
  })),
}));

describe('StudentDetail', () => {
  it('should render successfully', () => {
    const { container } = render(<StudentDetail />);

    expect(container).toMatchSnapshot();
  });
});
