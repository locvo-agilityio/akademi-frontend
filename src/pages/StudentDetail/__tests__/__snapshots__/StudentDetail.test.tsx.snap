// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentDetail should render successfully 1`] = `
<div>
  <div
    class="chakra-stack css-vwmgls"
  >
    <div
      class="css-13rbvqk"
    >
      <div
        class="css-tkevr6"
      >
        <div
          class="chakra-spinner css-1y7joxr"
          title="Fallback"
        >
          <span
            class="css-8b45rq"
          >
            Loading...
          </span>
        </div>
      </div>
      <div
        class="chakra-stack css-1p38if7"
      >
        <h3
          class="chakra-text css-0"
        >
          Payment History
        </h3>
        <div
          class="css-ab2388"
        >
          <div
            class="chakra-table__container css-cvn5rz"
          >
            <table
              class="chakra-table css-0"
            >
              <tbody
                class="css-0"
              >
                <tr
                  class="css-0"
                >
                  <td
                    class="css-1g7ry2m"
                  >
                    <div
                      class="css-19rjtd3"
                    >
                      <div
                        class="css-12482ex"
                      >
                        <svg
                          fill="none"
                          height="16"
                          viewBox="0 0 24 16"
                          width="24"
                        >
                          <path
                            d="M24 1.33333C24.0126 1.24043 24.0126 1.14624 24 1.05333C23.9884 0.975052 23.966 0.898765 23.9334 0.826666C23.8981 0.761503 23.858 0.699103 23.8134 0.64C23.7626 0.555671 23.6997 0.479292 23.6267 0.413333L23.4667 0.32C23.3897 0.262609 23.3042 0.21762 23.2134 0.186666H22.9467C22.8654 0.107993 22.7706 0.0447434 22.6667 0H16C15.6464 0 15.3073 0.140476 15.0572 0.390525C14.8072 0.640573 14.6667 0.979711 14.6667 1.33333C14.6667 1.68696 14.8072 2.02609 15.0572 2.27614C15.3073 2.52619 15.6464 2.66667 16 2.66667H19.7734L14.44 8.94667L8.68003 5.52C8.4072 5.35773 8.08544 5.29813 7.77259 5.3519C7.45975 5.40567 7.17635 5.56929 6.97336 5.81333L0.306694 13.8133C0.194425 13.9481 0.109838 14.1036 0.0577791 14.2711C0.00571983 14.4386 -0.0127892 14.6147 0.00331194 14.7893C0.0194131 14.9639 0.0698082 15.1337 0.151611 15.2888C0.233413 15.4439 0.345016 15.5814 0.480028 15.6933C0.71991 15.8921 1.02182 16.0006 1.33336 16C1.52924 16.0003 1.72279 15.9575 1.90023 15.8745C2.07768 15.7915 2.23466 15.6705 2.36003 15.52L8.29336 8.4L13.9867 11.8133C14.2567 11.9735 14.5746 12.0332 14.8844 11.982C15.1942 11.9308 15.4759 11.7719 15.68 11.5333L21.3334 4.93333V8C21.3334 8.35362 21.4738 8.69276 21.7239 8.94281C21.9739 9.19286 22.3131 9.33333 22.6667 9.33333C23.0203 9.33333 23.3595 9.19286 23.6095 8.94281C23.8596 8.69276 24 8.35362 24 8V1.33333Z"
                            fill="white"
                          />
                        </svg>
                      </div>
                      #
                      1
                    </div>
                  </td>
                  <td
                    class="css-1g7ry2m"
                  >
                    <p
                      class="chakra-text css-cif1vy"
                    >
                      February 10, 2025
                    </p>
                  </td>
                  <td
                    class="css-1g7ry2m"
                  >
                    <p
                      class="chakra-text css-19fbeh3"
                    >
                      $ 
                      100,000
                    </p>
                  </td>
                  <td
                    class="css-1g7ry2m"
                  >
                    <p
                      class="chakra-text css-1870vs7"
                    >
                      Pending
                    </p>
                  </td>
                </tr>
                <tr
                  class="css-0"
                >
                  <td
                    class="css-1g7ry2m"
                  >
                    <div
                      class="css-19rjtd3"
                    >
                      <div
                        class="css-12482ex"
                      >
                        <svg
                          fill="none"
                          height="16"
                          viewBox="0 0 24 16"
                          width="24"
                        >
                          <path
                            d="M24 1.33333C24.0126 1.24043 24.0126 1.14624 24 1.05333C23.9884 0.975052 23.966 0.898765 23.9334 0.826666C23.8981 0.761503 23.858 0.699103 23.8134 0.64C23.7626 0.555671 23.6997 0.479292 23.6267 0.413333L23.4667 0.32C23.3897 0.262609 23.3042 0.21762 23.2134 0.186666H22.9467C22.8654 0.107993 22.7706 0.0447434 22.6667 0H16C15.6464 0 15.3073 0.140476 15.0572 0.390525C14.8072 0.640573 14.6667 0.979711 14.6667 1.33333C14.6667 1.68696 14.8072 2.02609 15.0572 2.27614C15.3073 2.52619 15.6464 2.66667 16 2.66667H19.7734L14.44 8.94667L8.68003 5.52C8.4072 5.35773 8.08544 5.29813 7.77259 5.3519C7.45975 5.40567 7.17635 5.56929 6.97336 5.81333L0.306694 13.8133C0.194425 13.9481 0.109838 14.1036 0.0577791 14.2711C0.00571983 14.4386 -0.0127892 14.6147 0.00331194 14.7893C0.0194131 14.9639 0.0698082 15.1337 0.151611 15.2888C0.233413 15.4439 0.345016 15.5814 0.480028 15.6933C0.71991 15.8921 1.02182 16.0006 1.33336 16C1.52924 16.0003 1.72279 15.9575 1.90023 15.8745C2.07768 15.7915 2.23466 15.6705 2.36003 15.52L8.29336 8.4L13.9867 11.8133C14.2567 11.9735 14.5746 12.0332 14.8844 11.982C15.1942 11.9308 15.4759 11.7719 15.68 11.5333L21.3334 4.93333V8C21.3334 8.35362 21.4738 8.69276 21.7239 8.94281C21.9739 9.19286 22.3131 9.33333 22.6667 9.33333C23.0203 9.33333 23.3595 9.19286 23.6095 8.94281C23.8596 8.69276 24 8.35362 24 8V1.33333Z"
                            fill="white"
                          />
                        </svg>
                      </div>
                      #
                      2
                    </div>
                  </td>
                  <td
                    class="css-1g7ry2m"
                  >
                    <p
                      class="chakra-text css-cif1vy"
                    >
                      February 10, 2025
                    </p>
                  </td>
                  <td
                    class="css-1g7ry2m"
                  >
                    <p
                      class="chakra-text css-19fbeh3"
                    >
                      $ 
                      100,000
                    </p>
                  </td>
                  <td
                    class="css-1g7ry2m"
                  >
                    <p
                      class="chakra-text css-1j9enpg"
                    >
                      Complete
                    </p>
                  </td>
                </tr>
                <tr
                  class="css-0"
                >
                  <td
                    class="css-1g7ry2m"
                  >
                    <div
                      class="css-19rjtd3"
                    >
                      <div
                        class="css-12482ex"
                      >
                        <svg
                          fill="none"
                          height="16"
                          viewBox="0 0 24 16"
                          width="24"
                        >
                          <path
                            d="M24 1.33333C24.0126 1.24043 24.0126 1.14624 24 1.05333C23.9884 0.975052 23.966 0.898765 23.9334 0.826666C23.8981 0.761503 23.858 0.699103 23.8134 0.64C23.7626 0.555671 23.6997 0.479292 23.6267 0.413333L23.4667 0.32C23.3897 0.262609 23.3042 0.21762 23.2134 0.186666H22.9467C22.8654 0.107993 22.7706 0.0447434 22.6667 0H16C15.6464 0 15.3073 0.140476 15.0572 0.390525C14.8072 0.640573 14.6667 0.979711 14.6667 1.33333C14.6667 1.68696 14.8072 2.02609 15.0572 2.27614C15.3073 2.52619 15.6464 2.66667 16 2.66667H19.7734L14.44 8.94667L8.68003 5.52C8.4072 5.35773 8.08544 5.29813 7.77259 5.3519C7.45975 5.40567 7.17635 5.56929 6.97336 5.81333L0.306694 13.8133C0.194425 13.9481 0.109838 14.1036 0.0577791 14.2711C0.00571983 14.4386 -0.0127892 14.6147 0.00331194 14.7893C0.0194131 14.9639 0.0698082 15.1337 0.151611 15.2888C0.233413 15.4439 0.345016 15.5814 0.480028 15.6933C0.71991 15.8921 1.02182 16.0006 1.33336 16C1.52924 16.0003 1.72279 15.9575 1.90023 15.8745C2.07768 15.7915 2.23466 15.6705 2.36003 15.52L8.29336 8.4L13.9867 11.8133C14.2567 11.9735 14.5746 12.0332 14.8844 11.982C15.1942 11.9308 15.4759 11.7719 15.68 11.5333L21.3334 4.93333V8C21.3334 8.35362 21.4738 8.69276 21.7239 8.94281C21.9739 9.19286 22.3131 9.33333 22.6667 9.33333C23.0203 9.33333 23.3595 9.19286 23.6095 8.94281C23.8596 8.69276 24 8.35362 24 8V1.33333Z"
                            fill="white"
                          />
                        </svg>
                      </div>
                      #
                      3
                    </div>
                  </td>
                  <td
                    class="css-1g7ry2m"
                  >
                    <p
                      class="chakra-text css-cif1vy"
                    >
                      February 10, 2025
                    </p>
                  </td>
                  <td
                    class="css-1g7ry2m"
                  >
                    <p
                      class="chakra-text css-19fbeh3"
                    >
                      $ 
                      100,000
                    </p>
                  </td>
                  <td
                    class="css-1g7ry2m"
                  >
                    <p
                      class="chakra-text css-bqcsif"
                    >
                      Canceled
                    </p>
                  </td>
                </tr>
                <tr
                  class="css-0"
                >
                  <td
                    class="css-1g7ry2m"
                  >
                    <div
                      class="css-19rjtd3"
                    >
                      <div
                        class="css-12482ex"
                      >
                        <svg
                          fill="none"
                          height="16"
                          viewBox="0 0 24 16"
                          width="24"
                        >
                          <path
                            d="M24 1.33333C24.0126 1.24043 24.0126 1.14624 24 1.05333C23.9884 0.975052 23.966 0.898765 23.9334 0.826666C23.8981 0.761503 23.858 0.699103 23.8134 0.64C23.7626 0.555671 23.6997 0.479292 23.6267 0.413333L23.4667 0.32C23.3897 0.262609 23.3042 0.21762 23.2134 0.186666H22.9467C22.8654 0.107993 22.7706 0.0447434 22.6667 0H16C15.6464 0 15.3073 0.140476 15.0572 0.390525C14.8072 0.640573 14.6667 0.979711 14.6667 1.33333C14.6667 1.68696 14.8072 2.02609 15.0572 2.27614C15.3073 2.52619 15.6464 2.66667 16 2.66667H19.7734L14.44 8.94667L8.68003 5.52C8.4072 5.35773 8.08544 5.29813 7.77259 5.3519C7.45975 5.40567 7.17635 5.56929 6.97336 5.81333L0.306694 13.8133C0.194425 13.9481 0.109838 14.1036 0.0577791 14.2711C0.00571983 14.4386 -0.0127892 14.6147 0.00331194 14.7893C0.0194131 14.9639 0.0698082 15.1337 0.151611 15.2888C0.233413 15.4439 0.345016 15.5814 0.480028 15.6933C0.71991 15.8921 1.02182 16.0006 1.33336 16C1.52924 16.0003 1.72279 15.9575 1.90023 15.8745C2.07768 15.7915 2.23466 15.6705 2.36003 15.52L8.29336 8.4L13.9867 11.8133C14.2567 11.9735 14.5746 12.0332 14.8844 11.982C15.1942 11.9308 15.4759 11.7719 15.68 11.5333L21.3334 4.93333V8C21.3334 8.35362 21.4738 8.69276 21.7239 8.94281C21.9739 9.19286 22.3131 9.33333 22.6667 9.33333C23.0203 9.33333 23.3595 9.19286 23.6095 8.94281C23.8596 8.69276 24 8.35362 24 8V1.33333Z"
                            fill="white"
                          />
                        </svg>
                      </div>
                      #
                      4
                    </div>
                  </td>
                  <td
                    class="css-1g7ry2m"
                  >
                    <p
                      class="chakra-text css-cif1vy"
                    >
                      February 10, 2025
                    </p>
                  </td>
                  <td
                    class="css-1g7ry2m"
                  >
                    <p
                      class="chakra-text css-19fbeh3"
                    >
                      $ 
                      100,000
                    </p>
                  </td>
                  <td
                    class="css-1g7ry2m"
                  >
                    <p
                      class="chakra-text css-1j9enpg"
                    >
                      Complete
                    </p>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="css-4v9c5a"
        >
          <div
            class="css-13mwgsm"
          >
            <div
              class="css-70qvj9"
            >
              <p
                class="chakra-text css-ghb6lk"
              >
                30 items
              </p>
            </div>
            <div
              class="css-69i1ev"
            >
              <button
                class="chakra-button css-1umnp0x"
                disabled=""
                title="prev-button"
                type="button"
              >
                <svg
                  class="chakra-icon css-1r59igl"
                  fill="none"
                  focusable="false"
                  height="14"
                  viewBox="0 0 21 14"
                  width="21"
                >
                  <path
                    d="M20.6761 2.48803L11.2361 12.952C10.8521 13.4 10.1481 13.4 9.76415 12.952L0.324149 2.48803C-0.283851 1.81603 0.164149 0.728027 1.06015 0.728027L19.9401 0.728027C20.8361 0.728027 21.2841 1.81603 20.6761 2.48803Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
              <div
                class="css-j2at52"
              >
                <button
                  class="chakra-button css-18o81ng"
                  disabled=""
                  title="page-1"
                  type="button"
                >
                  1
                </button>
                <button
                  class="chakra-button css-c1bazq"
                  title="page-2"
                  type="button"
                >
                  2
                </button>
              </div>
              <button
                class="chakra-button css-1umnp0x"
                title="next-button"
                type="button"
              >
                <svg
                  class="chakra-icon css-178y3kn"
                  fill="none"
                  focusable="false"
                  height="14"
                  viewBox="0 0 21 14"
                  width="21"
                >
                  <path
                    d="M20.6761 2.48803L11.2361 12.952C10.8521 13.4 10.1481 13.4 9.76415 12.952L0.324149 2.48803C-0.283851 1.81603 0.164149 0.728027 1.06015 0.728027L19.9401 0.728027C20.8361 0.728027 21.2841 1.81603 20.6761 2.48803Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-tkevr6"
    >
      <div
        class="chakra-spinner css-1y7joxr"
        title="Fallback"
      >
        <span
          class="css-8b45rq"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
</div>
`;
